sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
I agree. Encoding state on the Txn proto itself just for client usage seems backwards and it looks too one-off to even have this on the proto (I don't see why you need it there, but haven't seen the whole diff just yet - so far looks like having it on `client.Txn` would be as good)